shared/services/rewards/generator.go

```diff
@@ -198,7 +198,7 @@ func NewTreeGenerator(logger *log.ColorLogger, logPrefix string, rp RewardsExecu
 			t.generatorImpl = info.generator
 			foundGenerator = true
 		}
-		if !foundApproximator && startInterval < t.index {
+		if !foundApproximator && (startInterval < t.index || t.index == 0) {
```

Comment:
Can we just make this `<=` like the generator now? Using the previous ruleset was always a bug. Update the comment on L189-89 too?